ansible/roles/machine/provision_multidomain/tasks/main.yml

```diff
@@ -0,0 +1,156 @@
+      set_fact:
+        dns_forwarder: "{{ dns_server_resolvectl.stdout }}"
+      when: dns_server_resolvectl is not skipped
+
+    # https://github.com/ansible/ansible/issues/56243
+    - name: ensure file already exists at template dest to work around 'invalid selinux context' issue
+      file:
+        path: "/vagrant/ipa-test-multidomain-config.yaml"
+        state: touch
+
+    - name: create test config file
+      template:
+        src: ipa-test-multidomain-config.yaml
+        dest: /vagrant/ipa-test-multidomain-config.yaml
+  when: inventory_hostname == 'controller' or deploy_ipa_test_config
+
+- name: add PR build repository
+  yum_repository:
+    baseurl: "{{ repofile_url | dirname }}"
+    priority: 1
+    name: freeipa-prci
+    description: FreeIPA PR CI testing packages
+    gpgcheck: no
+
+- name: "configure custom COPR repo ({{ copr }})"
+  shell: "dnf copr enable -y {{ copr }}"
+  when: copr is defined and copr
+
+- name: enable updates-testing repository
+  shell: "dnf config-manager --set-enabled updates-testing"
+  when: enable_testing_repo is defined and enable_testing_repo
+
+- name: update packages
+  dnf:
+    name: '*'
+    state: latest
+  when: update_packages is defined and update_packages
+
+# Workaround for pki upgrade issue
+- name: update pki packages
+  dnf:
+    name: 'dogtag-pki-server'
+    state: latest
+    allowerasing: true
+  when: update_packages is defined and update_packages
+
+- name: update pip packages
+  pip:
+    executable: pip3
+    name: "{{ python_packages_to_install }}"
+  when: update_packages is defined and update_packages
+
+- name: install freeipa packages
+  dnf:
+    state: latest
+    name:
+      - freeipa-*
+      - python*-ipatests
+    exclude:
+      - freeipa-fas
+  register: result
+  until: result.rc == 0
+  retries: 3
+  delay: 5
+
+- name: install client packages
+  dnf:
+    state: latest
+    name:
+    - samba-client
+  when: testing_ad is defined
+
+- name: install Fedora 27 specific tests dependencies
+  dnf:
+    state: latest
+    name:
+      - ntpdate
+      - sssd-tools
+  when: ansible_distribution == 'Fedora' and ansible_distribution_version == '27'
+
+- name: create directory to save installed packages logs
+  file:
+    path: /vagrant/installed_packages/
+    state: directory
+  # /vagrant is mounted using sshfs and there are random "operation not permitted" errors
+  register: pkgs_dir
+  until: pkgs_dir is not failed
+  retries: 3
+  delay: 10
+
+- name: get all packages
+  shell: rpm -qa | sort > /vagrant/installed_packages/installed_packages_{{inventory_hostname}}.log
+
+- name: create hosts file from template
+  template:
+    src: hosts
+    dest: /etc/hosts
+
+# workaround for https://github.com/ansible/ansible/issues/19814
+- name: set hostname
+  shell: "hostnamectl set-hostname {{ inventory_hostname }}.ipa.test"
+  when: inventory_hostname == 'controller' or inventory_hostname == 'ipamaster' or inventory_hostname == 'ipaclient'
```

Comment:
Instead of creating a new file, update the existing `ansible/roles/machine/provision/tasks/main.yml` with the following lines:
```
# workaround for https://github.com/ansible/ansible/issues/19814
- name: set hostname
  shell: "hostnamectl set-hostname {{ inventory_hostname }}.ipa.test"
  when: inventory_hostname == 'controller' or inventory_hostname == 'master' or inventory_hostname == 'client' or inventory_hostname == 'replica'

- name: set hostname
  shell: "hostnamectl set-hostname {{ inventory_hostname }}.trustedipa.test"
  when: inventory_hostname == 'controller' or inventory_hostname == 'trustedmaster' or inventory_hostname == 'trustedclient' or inventory_hostname == 'trustedreplica'
 ```